src/providers/data_provider/dp_target_id.c

```diff
@@ -592,19 +840,16 @@ static errno_t dp_get_account_info_initgroups_step(struct tevent_req *req)
                      state->initgr_ctx->groups);
         if (subreq == NULL) {
             DEBUG(SSSDBG_CRIT_FAILURE, "Unable to create subrequest!\n");
-            return ENOMEM;
+            tevent_req_error(req, ret);
         }
 
         tevent_req_set_callback(subreq, dp_get_account_info_done, req);
-        ret = EAGAIN;
     } else {
-        ret = EOK;
+        tevent_req_done(req);
```

Comment:
tevent_req_done must be the last call (move it after dp_req_initgr_pp_sr_overlay)

Reply:
Or perhaps move dp_req_initgr_pp_set_initgr* before the if? It is called unconditionally in the original code. It just should be more clear what is the intention.

Reply:
Fixed.